lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -77,31 +78,26 @@ sub _save_vars {
     close($fd);
 }
 
-# When changing something here, also take a look at OpenQA::Utils::asset_type_from_setting
 sub detect_asset_keys {
     my ($vars) = @_;
 
     my %res;
-    for my $isokey (qw(ISO), map { "ISO_$_" } (1 .. 9)) {
-        $res{$isokey} = 'iso' if $vars->{$isokey};
+    my $type;
+
+    for my $key (keys(%$vars)) {
+        # UEFI_PFLASH_VARS may point to an image uploaded by a previous
+        # test (which we should treat as an hdd asset), or it may point
+        # to an absolute filesystem location of e.g. a template file from
+        # edk2 (which we shouldn't).
+        next if $key eq 'UEFI_PFLASH_VARS' && $vars->{$key} =~ m,^/,;
+
+        # Skip external assets, they're handled elsewhere.
+        next if $vars->{$key . '_URL'} || $vars->{$key . '_DECOMPRESS_URL'};
+        $type = asset_type_from_setting($key, $vars->{$key});
+        next if $type eq 'repo' || !$type;
```

Comment:
We have `OpenQA::Schema::Result::Assets::is_type_hidden` which allows to distinguish "hidden" assets in general. Maybe it makes sense to use that function here as well (except this is really just about repos).

Reply:
Hidden assets still need to be cached on the worker so I can't use that function. I'm excluding repos because I'm not sure whether the cache service can handle caching whole directories or just single files. If it can handle directories, I'll change that line to `next if !$type;`

Reply:
Makes sense. Not sure whether the cache service can handle directories out of my head. Maybe @kraih knows.

Reply:
Just files, only the rsync part deals with directories so far.

Reply:
OK, I'll just put a FIXME in there and leave the condition as is then.

Reply:
I don't think we will ever implement that "TODO" or "FIXME". I recommend to remove it again as "TODOs" in code tend to stay forever. If you think we should have "repo caching" then please record that in a ticket instead.

Reply:
There's so many old TODO's already, i don't really pay much attention anymore. 😆 But @okurz is probably right.

Reply:
I sometimes find them helpful clues when reading code. Of course one has to be aware that it might be out dated. But usually I add comments like `# note: ...` when I want to explain a caveat or *why* the code has been written like that.